### PR TITLE
adding IfInBitRate, IfInPktRate, IfOutBitRate, IfOutPktRate to IF-MIB…

### LINKED ISSUE #404: FR: calculate additional metrics ifInBitRate, ifOutBitRate, ifInPacketRate and ifOutPacketRate for IF-MIB
... in the similar way we are doing it for IfInUtilization and IfOutUtilization

### PR DESCRIPTION
**Pull request by dpajin-kentik**
…::in measurement in Influx output
closes #404 

seems to be working:

```
IF-MIB::if,device_name=edge01.iad1,if_Alias=CORE:edge01-qfx\ 802.1q\ Trunk,if_Description=ae0,if_Index=564,if_MTU=1518,if_PhysAddress=5c:5e:ab:01:47:c0,if_Speed=40000,if_Type=ieee8023adLag,if_interface_name=ae0,instrumentation.name=juniper-mx-router,tags.dc=iad1,tags.kentik.model=MX80,tags.kentik.vendor=Juniper\ Networks,tags.site=iad,tags.type=edge ifInErrors=0i,ifHCInUcastPkts=61385352i,ifHCOutBroadcastPkts=0i,ifHCInOctets=53564747891i,ifOutDiscards=0i,ifHCOutOctets=53547600083i,ifOutErrorPercent=0i,ifHCOutMulticastPkts=0i,if_AdminStatus=1i,ifInErrorPercent=0i,ifInDiscards=0i,ifOutErrors=0i,ifHCOutUcastPkts=61360860i,if_OperStatus=1i,ifHCInBroadcastPkts=0i,ifHCInMulticastPkts=12311i,IfInBitRate=7.20319353047571e+09,IfInPktRate=1.0320669524289797e+06,IfOutUtilization=18.002218888216508,IfOutBitRate=7.200887555286603e+09,IfOutPktRate=1.0314483106404438e+06,IfInUtilization=18.007983826189275,if_OperStatus_str="up",if_AdminStatus_str="up" 1661466250000000000
```

### CODE REVIEW

**Review by tdanner — APPROVED** [review]
Looks ok to me. Probably good to wait for @i3149 's review before merging.

File: pkg/formats/influx/influx.go
Comment (i3149) [inline pkg/formats/influx/influx.go:585]: My only comment is to just use attrNew here for bitrate. No need to get a seperate map because whatever fields are valid for utilization will also be valid for bitrate (for a given interface).

File: pkg/formats/influx/influx.go
Comment (i3149) [inline pkg/formats/influx/influx.go:596]: Drop on filter will also be the same across here so maybe move all these metrics up into the common if statement for attrNew?

**Review by i3149 — COMMENTED** [review]
Looks like it should work but to make things a bit more readable how about abstracting out these `rate` metrics into a separate function and then re-using same attributes?

**Comment by dpajin-kentik** [comment]
Thanks @i3149, I have no clue about Go, but I know something about copy/paste :)

**Comment by i3149** [comment]
I did a pass on a refactor -- @tdanner what do you think?

**Review by tdanner — APPROVED** [review]
Looks ok to me

**Pull request merged**
